commit 952a6ac6911ee697117eaf0f8c42ff6582d351a6
Author: Julian Scheel <julian@jusst.de>
Date:   Thu Aug 14 08:42:55 2014 +0200

    lirc_rpi: Add parameter to specify input pin pull
    
    Depending on the connected IR circuitry it might be desirable to change the
    gpios internal pull from it pull-down default behaviour. Add a module
    parameter to allow the user to set it explicitly.
    
    Signed-off-by: Julian Scheel <julian@jusst.de>

diff --git a/drivers/staging/media/lirc/lirc_rpi.c b/drivers/staging/media/lirc/lirc_rpi.c
index 1da6a9f..887c36b 100644
--- a/drivers/staging/media/lirc/lirc_rpi.c
+++ b/drivers/staging/media/lirc/lirc_rpi.c
@@ -41,6 +41,8 @@
 #include <mach/gpio.h>
 #include <linux/gpio.h>
 
+#include <linux/platform_data/bcm2708.h>
+
 #define LIRC_DRIVER_NAME "lirc_rpi"
 #define RBUF_LEN 256
 #define LIRC_TRANSMITTER_LATENCY 50
@@ -62,6 +64,8 @@
 
 /* set the default GPIO input pin */
 static int gpio_in_pin = 18;
+/* set the default pull behaviour for input pin */
+static int gpio_in_pull = BCM2708_PULL_DOWN;
 /* set the default GPIO output pin */
 static int gpio_out_pin = 17;
 /* enable debugging messages */
@@ -318,6 +322,7 @@ static int init_port(void)
 		goto exit_gpio_free_out_pin;
 	}
 
+	bcm2708_gpio_setpull(gpiochip, gpio_in_pin, gpio_in_pull);
 	gpiochip->direction_input(gpiochip, gpio_in_pin);
 	gpiochip->direction_output(gpiochip, gpio_out_pin, 1);
 	gpiochip->set(gpiochip, gpio_out_pin, invert);
@@ -666,6 +671,10 @@ module_param(gpio_in_pin, int, S_IRUGO);
 MODULE_PARM_DESC(gpio_in_pin, "GPIO input pin number of the BCM processor."
 		 " (default 18");
 
+module_param(gpio_in_pull, int, S_IRUGO);
+MODULE_PARM_DESC(gpio_in_pull, "GPIO input pin pull configuration."
+		 " (0 = off, 1 = up, 2 = down, default down)");
+
 module_param(sense, int, S_IRUGO);
 MODULE_PARM_DESC(sense, "Override autodetection of IR receiver circuit"
 		 " (0 = active high, 1 = active low )");
